app/controllers/bulk_submissions_controller.rb

```diff
@@ -22,6 +22,14 @@ def destroy
     redirect_to authenticated_root_path
   end
 
+  def download
+    bulk_submission = BulkSubmission.find(params[:id])
+    Rails.logger.info "User #{current_user.id} downloaded results file for bulk submission #{bulk_submission.id}"
+    attachment = bulk_submission.result_file.attachment
+    data = attachment.blob.download
+    send_data data, filename: attachment.filename.to_s, content_type: 'text/csv'
```

Comment:
happy to go with this but have concerns over performance due to file sizes (but should be alright) that end up in RAM. I would not bother with the interim `data` var though

```suggestion
    attachment = bulk_submission.result_file.attachment
    send_data attachment.blob.download, 
      filename: attachment.filename.to_s, 
      content_type: 'text/csv'
```

Reply:
👍 have made suggested change. My understanding is that if we want to avoid a redirect we have to read into RAM so will leave as is for now and  monitor for errors.